tasks/main.yml

```diff
@@ -46,96 +82,139 @@
+      if not kernel_settings_purge
+      and kernel_settings_sysfs != __kernel_settings_state_empty
+      and not __kernel_settings_previous_replaced in kernel_settings_sysfs
+      else {} }}"
+    __systemd_old: "{{
+      __kernel_settings_profile_contents.data.get('systemd', {}).get('cpu_affinity')
+      if not kernel_settings_purge
+      and kernel_settings_systemd_cpu_affinity != __kernel_settings_state_absent
+      else none }}"
+    __trans_huge_old: "{{
+      __kernel_settings_profile_contents.data.get('vm', {}).get('transparent_hugepages')
+      if not kernel_settings_purge
+      and kernel_settings_transparent_hugepages != __kernel_settings_state_absent
+      else none }}"
+    __trans_defrag_old: "{{
+      __kernel_settings_profile_contents.data.get('vm', {}).get('transparent_hugepage.defrag')
+      if not kernel_settings_purge
+      and kernel_settings_transparent_hugepages_defrag != __kernel_settings_state_absent
+      else none }}"
+    __sections:
+      - name: sysctl
+        new: "{{ kernel_settings_sysctl | difference([__kernel_settings_previous_replaced]) | list
+          if kernel_settings_sysctl != __kernel_settings_state_empty
+          else [] }}"
+        old: "{{ __sysctl_old }}"
+      - name: sysfs
+        new: "{{ kernel_settings_sysfs | difference([__kernel_settings_previous_replaced]) | list
+          if kernel_settings_sysfs != __kernel_settings_state_empty
+          else [] }}"
+        old: "{{ __sysfs_old }}"
+      - name: systemd
+        new:
+          - name: cpu_affinity
+            value: "{{ kernel_settings_systemd_cpu_affinity }}"
+        old:
+          cpu_affinity: "{{ __systemd_old }}"
+      - name: vm
+        new:
+          - name: transparent_hugepages
+            value: "{{ kernel_settings_transparent_hugepages }}"
+          - name: transparent_hugepage.defrag
+            value: "{{ kernel_settings_transparent_hugepages_defrag }}"
+        old:
+          transparent_hugepages: "{{ __trans_huge_old }}"
+          transparent_hugepage.defrag: "{{ __trans_defrag_old }}"
+  register: __kernel_settings_register_apply
+
+# this will also apply the kernel_settings profile, so we
+# can skip the apply profile step in this case
+- name: Restart tuned to apply active profile, mode changes
+  service:
+    name: "{{ item }}"
+    state: restarted
+    enabled: true
+  loop: "{{ __kernel_settings_services }}"
+  when: __kernel_settings_register_profile is changed or
+    __kernel_settings_register_mode is changed
 
 - name: Tuned apply settings
-  command: >
-    tuned-adm profile '{{ __kernel_settings_register_module.active_profile }}'
-  when: __kernel_settings_register_module is changed  # noqa no-handler
+  command: >-
+    tuned-adm profile {{ __kernel_settings_active_profile | quote }}
+  when:
+    - not __kernel_settings_register_profile is changed
+    - not __kernel_settings_register_mode is changed
```

Comment:
In the above task, there is `or` between these two checks, here they both should be `changed` to restart apply tuned?